### PR TITLE
Adds the possiblity to use replacement tags in a TrackingScript

### PR DESCRIPTION
**Pull request by TRoZZNL**
An implementation of IScriptTagContentResolver can be added to a web project to enable page / url-dependant replacement values which will be replaced in the eventual output of the TrackingScript code.

To use this new feature, add the following code to the startup class :
`services.AddTransient<IScriptTagContentResolver, ScriptTagContentResolver>();`
After 
`services.AddMediaKiwiContentService();`

Where **ScriptTagContentResolver** is the class implementing the **IScriptTagContentResolver** interface

### CODE REVIEW

File: src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs [hunk 1]
```diff
@@ -19,6 +19,40 @@ public class ScriptTagHelperComponent : TagHelperComponent
         [ViewContext]
         public ViewContext ViewContext { get; set; }
 
+        private void PerformReplacements(TrackingScriptSimple script)
+        {
+            if (script?.ReplacementItems?.Count > 0)
+            {
+                // Are all marked as mandatory ?
+                var allMandatory = script.ReplacementItems.Count == script.ReplacementItems.Count(x => x.MustBeFilled);
```
Comment (bart-van-der-drift) [inline src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs:27]: Use script.ReplacementItems.All(x => x.MustBeFilled) for better performance

File: src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs [hunk 2]
```diff
@@ -19,6 +19,40 @@ public class ScriptTagHelperComponent : TagHelperComponent
...
+
+                // are all values filled ?
+                var allFilled = script.ReplacementItems.Count == script.ReplacementItems.Count(x => string.IsNullOrWhiteSpace(x.Value) == false);
```
Comment (bart-van-der-drift) [inline src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs:30]: Use script.ReplacementItems.All(x => string.IsNullOrWhiteSpace(x.Value) == false) for better performance

**Review by bart-van-der-drift — CHANGES_REQUESTED** [review]
Please see comments

File: src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs (hunk 1 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs]: Agreed, done

File: src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs (hunk 2 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi.Headless/ScriptTagHelperComponent.cs]: Agreed (though it should be All, not Any), done

**Review by bart-van-der-drift — APPROVED** [review]

**Pull request merged**